js/runner_test.go

```diff
@@ -548,15 +549,27 @@ func TestVURunInterrupt(t *testing.T) {
 
 	testdata := map[string]*Runner{"Source": r1, "Archive": r2}
 	for name, r := range testdata {
+		name, r := name, r
 		t.Run(name, func(t *testing.T) {
-			vu, err := r.newVU(make(chan stats.SampleContainer, 100))
+			ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
+			defer cancel()
+			samples := make(chan stats.SampleContainer, 100)
+			go func() {
+				select {
+				case <-ctx.Done():
```

Comment:
This doesn't seem like a good idea ... the context can be *done* and there could still be things to be written to `samples` ...
 I think you should do as I did at in 36f24852eee0806c756b3651659de4313d37b838 as that way it is even less likely that we will get this to block. 
Ideally we should have a function that returns the channel, start reading it, and we just need to close the channel when we are done.

Reply:
Hmm good point, I'll fix it. Though I don't think that `samples` channel will ever be closed, and using `PushIfNotCancelled()` won't fill the channel too much after the context is closed, since this is a test, the infinite `range` would still be better than this more complicated "fix"

Reply:
Fixed

Reply:
you could've written `defer close(samples)` somewhere ...

Reply:
maybe, but I'm not sure this wouldn't result in a panic for writing to a closed channel...

Reply:
I don't think it should be possible and if it happens than ... it is a bug that we need to fix